source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2397 @@
+        data: {
+          failCommands: [<string>, ...],
+          closeConnection: <boolean>,
+          errorCode: <integer>,
+          writeConcernError: <document>,
+          appName: <string>,
+          blockConnection: <boolean>,
+          blockTimeMS: <integer>,
+        }
+    });
+
+``failCommand`` supports the following ``data`` options, which may be combined
+if desired:
+
+* ``failCommands``: Required array of strings. Lists the command names to fail.
+* ``closeConnection``: Optional boolean, which defaults to ``false``. If
+  ``true``, the command will not be executed, the connection will be closed, and
+  the client will see a network error.
+* ``errorCode``: Optional integer, which is unset by default. If set, the
+  command will not be executed and the specified command error code will be
+  returned as a command error.
+* ``appName``: Optional string, which is unset by default. If set, the fail
+  point will only apply to connections for MongoClients created with this
+  ``appname``. New in server 4.4.0-rc2
+  (`SERVER-47195 <https://jira.mongodb.org/browse/SERVER-47195>`_).
+* ``blockConnection``: Optional boolean, which defaults to ``false``. If
+  ``true``, the server should block the affected commands for ``blockTimeMS``.
+  New in server 4.3.4
+  (`SERVER-41070 <https://jira.mongodb.org/browse/SERVER-41070>`_).
+* ``blockTimeMS``: Optional integer, which is required when ``blockConnection``
+  is ``true``. The number of milliseconds for which the affected commands should
+  be blocked. New in server 4.3.4
+  (`SERVER-41070 <https://jira.mongodb.org/browse/SERVER-41070>`_).
+
+
+Determining if a Sharded Cluster Uses Replica Sets
+--------------------------------------------------
+
+When connected to a mongos server, the test runner can query the
+`config.shards <https://docs.mongodb.com/manual/reference/config-database/#config.shards>`__
+collection. Each shard in the cluster is represented by a document in this
+collection. If the shard is backed by a single server, the ``host`` field will
+contain a single host. If the shard is backed by a replica set, the ``host``
+field contain the name of the replica followed by a forward slash and a
+comma-delimited list of hosts.
+
+
+Design Rationale
+================
+
+This specification was primarily derived from the test formats used by the
+`Transactions <../transactions/transactions.rst>`__ and
+`CRUD <../crud/crud.rst>`__ specs, which have served models or other specs.
+
+
+Breaking Changes
+================
+
+This section is reserved for future use. Any breaking changes to the test format
+should be described here in detail for historical reference, in addition to any
+shorter description that may be added to the `Change Log`_.
+
+
+Open Questions
+==============
+
+Note: these questions should be resolved and this section removed before
+publishing version 1.0 of the spec.
+
+
+General
+-------
+
+
+Interaction between top-level and test-level runOn
```

Comment:
I favor having more restrictive test leve runOns for easy readability and implementation.

Reply:
Sounds like we're on the same page about preferring more restrictive test-level runOns. I'll leave this open until we meet Wednesday to finalize team review.